Some bugfixes

SimoneSchmidt opened the pull request:
- Added Extended Metadata Page to Resource Wizards
- Fixed feedback-link in footer
- Added check in Delete Series Dialog
- Adapted source option visibility
- Adapted capture agent drop-downs
- Fixed Bulk Schedule Update

SimoneSchmidt committed:
Added Extended Metadata Page to Resource Wizards

Added Extended Metadata Page to New Event and New Series Wizards.
Added Extended Metadata section to Wizard Summary pages.
Adapted eventThunks and seriesThunks, to get extended metadata info out of API request and to add extended metadata values to API requests for creating new resources.
Removed some unnecesarry imports.

SimoneSchmidt committed:
Merge branch 'admin-ui-picard' into picard-bugfixes

SimoneSchmidt committed:
Fixed feedback-link in footer

Url value for the feedback-link in the site-footer is now the value from the org.properties from the user info, fetched via API.

SimoneSchmidt committed:
Added check in Delete Series Dialog

Added checks for:
    - does the series contain events
    - is deletion of series containing events allowed
to opening of Delete Series Dialog.
If deletion is allowed and the series contains events, a warning is now displayed.
If deletion is not allowed an error message is now displayed and deletion is not possible (no delete button).

SimoneSchmidt committed:
Adapted source option visibility

In source page of New Event Wizard,  source option visibility was adapted. Selecting the source modes "SCHEDULE_SINGLE" and "SCHEDULE_MULTIPLE" is now only possible and they are only shown, if capture agents exist and the user has the corresponding access role(s) to use (at least one of) them.

SimoneSchmidt committed:
Adapted capture agent drop-downs

In new event wizard, edit scheduled events modal and event details modal, the capture agent drop-downs have been adapted. Now only capture agents for which the user has access rights are available for selection in the drop-downs.
Fixed event selection in edit scheduled events modal, so that the user has to have access rights for the used capture agent in order to be able to select it.
Some slight adjustments of source option visibility in new event wizard.
Fixed empty available options in some drop-downs, they are now gone.
Slight refactoring.

isabellakutger commented on modules/admin-ui-picard/app/src/actions/eventActions.js:
```diff
@@ -55,9 +55,9 @@ export const loadEventMetadataInProgress = () => ({
 });
 
-export const loadEventMetadataSuccess = metadata => ({
+export const loadEventMetadataSuccess = ( usualMetadata, extendedMetadata ) => ({
```
keep the name `metadata` for standard metadata. I think nowhere else this is called `usualMetadata`

isabellakutger commented on modules/admin-ui-picard/app/src/actions/eventActions.js:
```diff
@@ -55,9 +55,9 @@ export const loadEventMetadataInProgress = () => ({
+export const loadEventMetadataSuccess = ( usualMetadata, extendedMetadata ) => ({
     type: LOAD_EVENT_METADATA_SUCCESS,
-    payload: { metadata }
+    payload: { usualMetadata, extendedMetadata }
```
See comment before

isabellakutger commented on modules/admin-ui-picard/app/src/components/events/partials/ModalTabsAndPages/DetailsExtendedMetadataTab.js:
```diff
@@ -29,7 +29,13 @@ const DetailsExtendedMetadataTab = ({ resourceId, editAccessRole, metadata, upda
+                    value = true;
+                } else if (value === 'false') {
+                    value = false;
+                }
```
This could be also used in other places, so maybe move this to an utils file

isabellakutger commented on modules/admin-ui-picard/app/src/components/events/partials/ModalTabsAndPages/EditScheduledEventsSummaryPage.js:
```diff
@@ -93,6 +101,22 @@ const EditScheduledEventsSummaryPage = ({ previousPage, formik }) => {
+            }
+        }
+        return false;
+    };
```
Move to utils file

isabellakutger commented on modules/admin-ui-picard/app/src/components/events/partials/ModalTabsAndPages/EventDetailsSchedulingTab.js:
```diff
@@ -307,7 +302,7 @@ const EventDetailsSchedulingTab = ({ eventId, t,
             scheduleEndMinute: makeTwoDigits(source.end.minute),
             captureAgent: source.device.name,
-            inputs: Array.from(source.device.inputMethods)
+            inputs: !!source.device.inputMethods? Array.from(source.device.inputMethods) : []
```
Space before '?'

isabellakutger commented on modules/admin-ui-picard/app/src/components/events/partials/ModalTabsAndPages/NewMetadataExtendedPage.js:
```diff
@@ -4,58 +4,80 @@ import { Field } from "formik";
-            </>
-        );
+            {/* Button for navigation to next page */}
+            <WizardNavigationButtons nextPage={nextPage}
```
set `noValidation` flag

isabellakutger commented on modules/admin-ui-picard/app/src/components/shared/ConfirmModal.js:
```diff
@@ -22,19 +23,45 @@ const ConfirmModal = ({ close, resourceType, resourceName, resourceId, deleteMet
+                {deleteAllowed ? (
+                    <div>
+                        {showCautionMessage && (
+                            <div ng-if="hasEvents && deleteSeriesWithEventsAllowed" className="modal-alert warning">
```
Delete the `ng-if` prop.

isabellakutger commented on modules/admin-ui-picard/app/src/reducers/eventReducers.js:
```diff
@@ -108,17 +109,19 @@ const events = (state=initialState, action) => {
         }
         case LOAD_EVENT_METADATA_SUCCESS: {
-            const { metadata } = payload;
+            const { usualMetadata, extendedMetadata } = payload;
```
Change name for standard metadata back to `metadata`

isabellakutger commented on modules/admin-ui-picard/app/src/reducers/eventReducers.js:
```diff
@@ -108,17 +109,19 @@ const events = (state=initialState, action) => {
                 ...state,
                 isLoading: false,
-                metadata: metadata
+                metadata: usualMetadata,
```
see comment before

isabellakutger commented on modules/admin-ui-picard/app/src/thunks/eventThunks.js:
```diff
@@ -74,9 +75,19 @@ export const fetchEventMetadata = () => async dispatch => {
 
-        const metadata = transformMetadataCollection(response[0]);
+        const mainCatalog = 'dublincore/episode';
+        let usualMetadata = {};
```
change name of standard metadata back to `metadata`

isabellakutger commented on modules/admin-ui-picard/app/src/thunks/seriesThunks.js:
```diff
@@ -48,9 +51,19 @@ export const fetchSeriesMetadata = () => async dispatch => {
 
-        const metadata = transformMetadataCollection(response[0]);
+        const mainCatalog = 'dublincore/series';
+        let usualMetadata = {};
```
change name back to `metadata`

isabellakutger commented on modules/admin-ui-picard/app/src/utils/resourceUtils.js:
```diff
@@ -330,6 +408,39 @@ export const transformAclTemplatesResponse = acl => {
+    } else {
+        const devicesWithAccessRights = [];
+        for(const device of inputDevices){
+            console.log(device)
```
remove log

isabellakutger reviewed (changes requested):
- Extended Metadata in New Event Wizard: Auto focus of the first field is not on the first field of the page but on the first field of the last table
- Error message in series deletion modal has no translation

SimoneSchmidt committed:
Fixed Bulk Schedule Update

On the edit page of the udate scheduled event modal, the series dropdown and the weekday selection have been fixed.
On the summary page of the modal, the way series name and weekday are displayed have been fixed.
The request data for the API request for updating the event schedules has been adapted, so that the metadata and schedule are properly updated.

SimoneSchmidt committed:
Changes according to first review of PR #537

isabellakutger commented on modules/admin-ui-picard/app/src/thunks/eventThunks.js:
```diff
@@ -3,14 +3,15 @@ import axios from "axios";
     loadEventMetadataFailure,
     loadEventMetadataInProgress,
-    loadEventMetadataSuccess,
+    loadEventMetadataSuccess, loadEventSchedulingFailure, loadEventSchedulingInProgress, loadEventSchedulingSuccess,
```
each in its own line

SimoneSchmidt committed:
Changes according to second review of PR #537

SimoneSchmidt committed:
Merge branch 'admin-ui-picard' into picard-bugfixes
